### PR TITLE
feat: profile edit be

### PR DESCRIPTION
**Pull request by ice-orion**
## Description
This PR adds BE integration for the edit profile

## Additional Notes
* Adds `UserMetadataNotifier`, `BannerPickerNotifier`, change `AvatarPickerNotifier`
* Fixes edit profile UI, add loading states
* Adds `useDraftMetadata` hook to handle draft user state 

## Type of Change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

## Screenshots 

https://github.com/user-attachments/assets/7af67b9a-631f-4bc0-a56a-850edf943f1b

### CODE REVIEW

**Review by ice-kreios — DISMISSED** [review]

File: lib/app/features/user/providers/avatar_picker_notifier.dart [hunk 1]
```diff
@@ -15,44 +16,49 @@ sealed class AvatarPickerState with _$AvatarPickerState {
...
+      state = AvatarPickerState.picked(file: MediaFile(path: assetImagePath));
       final croppedImage = await mediaService.cropImage(
         uiSettings: cropUiSettings,
-        path: cameraImagePath,
+        path: assetImagePath,
       );
-      if (croppedImage == null) return;
-      state = AvatarPickerState.picked(file: croppedImage);
+      if (croppedImage == null) {
+        state = const AvatarPickerState.initial();
```
Comment (ice-endymion) [inline lib/app/features/user/providers/avatar_picker_notifier.dart:51]: Maybe we can show some messages to the user? 
If cropping or compression fails, the state reverts to initial, but the user is not informed, potentially leading to confusion.

File: lib/app/features/user/providers/banner_picker_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      }
+      state = BannerPickerState.picked(file: MediaFile(path: assetImagePath));
+
+      final croppedImage = await mediaService.cropImage(
+        uiSettings: cropUiSettings,
+        path: assetImagePath,
+        aspectRatio: const CropAspectRatio(ratioX: 4, ratioY: 3),
+      );
+      if (croppedImage == null) {
+        state = const BannerPickerState.initial();
```
Comment (ice-endymion) [inline lib/app/features/user/providers/banner_picker_notifier.dart:54]: Also what about showing a message to the user about an error in cropping?

**Review by ice-endymion — CHANGES_REQUESTED** [review]

File: lib/app/features/auth/views/components/user_data_inputs/bio_input.dart [hunk 3]
```diff
@@ -6,16 +6,23 @@ import 'package:ion/app/features/auth/views/components/user_data_inputs/general_
 import 'package:ion/generated/assets.gen.dart';
 
 class BioInput extends StatelessWidget {
-  const BioInput({required this.controller, super.key});
+  const BioInput({this.controller, super.key, this.onChanged, this.initialValue});
```
Comment (ice-tychon) [inline lib/app/features/auth/views/components/user_data_inputs/bio_input.dart:9]: No trailing comma and `super.*` must be in the end.

File: lib/app/features/user/pages/profile_edit_page/components/edit_submit_button/edit_submit_button.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      type: hasChanges ? ButtonType.primary : ButtonType.disabled,
+      leadingIcon: ref.watch(userMetadataNotifierProvider).isLoading
+          ? const IONLoadingIndicator()
+          : Assets.svg.iconProfileSave.icon(
+              color: context.theme.appColors.onPrimaryAccent,
+            ),
+      onPressed: () async {
+        if (formKey.currentState!.validate()) {
+          final avatarFile =
+              ref.watch(avatarPickerNotifierProvider).whenOrNull(processed: (file) => file);
```
Comment (ice-tychon) [inline lib/app/features/user/pages/profile_edit_page/components/edit_submit_button/edit_submit_button.dart:45]: It should be `ref.read`, the same for `bannerFile`.

File: lib/app/features/user/providers/user_metadata_notifier.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          media: {...data.media, attachment.url: attachment},
+        );
+      }
+
+      await ref.read(nostrNotifierProvider.notifier).sendEntitiesData([...files, data]);
+    });
+  }
+
+  Future<(FileMetadata, MediaAttachment)?> _upload(MediaFile? file) {
+    return file != null
```
Comment (ice-tychon) [inline lib/app/features/user/providers/user_metadata_notifier.dart:51]: If you change return type `Future<({FileMetadata fileMeta, MediaAttachment attachment})?>` you'll be able to get rid of `uploadedAvatar.$2` above in the code.

File: lib/app/features/user/providers/user_metadata_notifier.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import 'package:ion/app/features/nostr/providers/nostr_notifier.dart';
+import 'package:ion/app/features/nostr/providers/nostr_upload_notifier.dart';
+import 'package:ion/app/features/user/model/user_metadata.dart';
+import 'package:ion/app/services/media_service/media_service.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'user_metadata_notifier.g.dart';
+
+@riverpod
+class UserMetadataNotifier extends _$UserMetadataNotifier {
```
Comment (ice-tychon) [inline lib/app/features/user/providers/user_metadata_notifier.dart:14]: Imho, the naming is poor in this file. `UserMetadataNotifier` and it's method `send` doesn't say what exactly they do. I'd name it something like `EditProfileNotifier`.

**Review by ice-endymion — DISMISSED** [review]
LGTM 👌

**Review by ice-tychon — DISMISSED** [review]

**Review by ice-endymion — APPROVED** [review]

**Pull request merged**